sharedbuffers/mapped_struct.py

```diff
@@ -2850,7 +2852,7 @@ def unpack_from(cls, buf, offs, idmap = None):
 
         packer = cls.PACKER
         timestamp, = packer.unpack_from(buf, offs)
-        rv =  date.fromtimestamp(timestamp)
+        rv =  date.fromtimestamp(time.mktime(time.gmtime(timestamp)))
```

Comment:
```suggestion
        rv =  datetime.utcfromtimestamp(timestamp).date()
```

That works reliably.

Reply:
Try to add a few tests

Reply:
Done!